src/dfx/Cargo.toml

```diff
@@ -44,7 +44,7 @@ garcon = { version = "0.2", features = ["async"] }
 hex = {version = "0.4.2", features = ["serde"] }
 humantime = "2.1.0"
 humantime-serde = "1.1.1"
-ic-types = "0.3.0"
+ic-types = "0.4.0"
```

Comment:
Could you check if we can remove direct dependency of `ic-types` here?
Since candid re-export `Pincipal` from `ic-types`

Reply:
yup, works :) 
the only thing I'm concerned about is the git diff size for this PR: this change encompasses +50 files https://github.com/dfinity/sdk/pull/2320/commits/fd8344e4fbf3a3c5700a6d8daf33ce40d96b024e

@ericswanson-dfinity let me know if you'd want to see this reverted and moved to separate PR